VideoPress: Add poster control

leogermani opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to projects/plugins/jetpack/to-test.md in a new commit as part of your PR. -->
<!-- a12s: If you have an expected version that you're aiming for the PR to add, please use the Milestone field to communicate it. If you leave it blank, that indicates there isn't a preference. -->

#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->
* Adds a control to manage a custom Poster image for the video

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->
NA

#### Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->
No

#### Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Add a new block and add a video (either by upload or URL)
* Check the new Poster image control in the Toolbar

![Captura de tela de 2022-07-01 17-47-48](https://user-images.githubusercontent.com/971483/176967705-026972f3-f6a0-44b1-a496-9a1bf7acef88.png)

* Test defining a custom poster image and confirm it works as expected. The preview image of the video should change
* Test opening the dialog again and removing the image

![Captura de tela de 2022-07-01 17-45-17](https://user-images.githubusercontent.com/971483/176967778-d40c7607-0939-4183-b65c-232f30380ff8.png)

EDIT - I wasn't happy with the result above, because
* We were using the full sized image to show the thumbnail, making it sometimes slow and ugly (browser force resizing)
* We can actually see the image just beneath the menu, in the actual video, so it was redundant.

Here is how it looks now:
![Captura de tela de 2022-07-06 17-24-40](https://user-images.githubusercontent.com/971483/177636706-56f31856-b982-4f6f-afd1-1c53c6b10705.png)




---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - 0-as-1202532622469874

leogermani committed:
start implementing poster control

leogermani committed:
update player and close modal on select

leogermani committed:
Merge branch 'trunk' into add/videopress-poster-control

leogermani committed:
implement in navigabe menu form

retrofox commented on projects/plugins/jetpack/extensions/blocks/videopress/v6/components/poster-image-block-control/style.module.scss:
```diff
@@ -0,0 +1,16 @@
+@import '@automattic/jetpack-base-styles/gutenberg-base-styles';
+
+// Hack to hide when the media modal is open
```
do you know whether Is it a known bug?

leogermani replied:
Initially I was fixing it with z-index, until I found this code here -> https://github.com/WordPress/gutenberg/blob/3198c904a982589ffcbc6481b8715dbf56fb2fd0/packages/block-editor/src/components/media-replace-flow/style.scss#L1

So yes, I think it`s a known issue

retrofox commented on projects/plugins/jetpack/extensions/blocks/videopress/v6/components/poster-image-block-control/index.js:
```diff
@@ -0,0 +1,119 @@
+		if ( poster ) {
+			return (
+				<>
+					<span>{ __( 'Current Image:', 'jetpack' ) }</span>
```
Pretty sure we agree we'd like to polish the UI here, in follow-up PRs. Anyway, just in case let me add the now :-)

retrofox reviewed (dismissed):
LGTM. Let me add some basic feedback

<img width="276" alt="image" src="https://user-images.githubusercontent.com/77539/177157221-714ef975-d380-4af0-b804-db7dd4606cca.png">

* Probably we'd like to change the icon :-)
* I'm not sure about showing the cover image in the toolbar. it's ok to be able to edit the poster from there, but maybe we should show the image in the inspector controls instead, something that we already do in the current implementation:
<img width="545" alt="Screen Shot 2022-07-04 at 9 45 46 AM" src="https://user-images.githubusercontent.com/77539/177157788-9068ff1e-8e82-4c91-a05d-9cb835a4b09a.png">

Anyway, LGTM. 🚀 !

leogermani commented:
> LGTM. Let me add some basic feedback
> 
> <img alt="image" width="276" src="https://user-images.githubusercontent.com/77539/177157221-714ef975-d380-4af0-b804-db7dd4606cca.png">
> 
>     * Probably we'd like to change the icon :-)
> 
>     * I'm not sure about showing the cover image in the toolbar. it's ok to be able to edit the poster from there, but maybe we should show the image in the inspector controls instead, something that we already do in the current implementation:
> 
> 
> <img alt="Screen Shot 2022-07-04 at 9 45 46 AM" width="545" src="https://user-images.githubusercontent.com/77539/177157788-9068ff1e-8e82-4c91-a05d-9cb835a4b09a.png">
> 
> Anyway, LGTM. rocket !

I'm gonna ask Design feedback for this before merging.

One thing I'm not liking very much is how we are displaying the current image. We are using full sized image in that small area, taking a while to load and being forced resized by the browser.. not looking very good.

> I'm not sure about showing the cover image in the toolbar. it's ok to be able to edit the poster from there, but maybe we should show the image in the inspector controls instead, something that we already do in the current implementation

This was requested by @oskosk . Not sure if there was any previous discussion about this.

retrofox commented:
> This was requested by @oskosk . Not sure if there was any previous discussion about this.

Let's ask designer folks, too.

leogermani commented:
As for the icon, maybe the regular image icon:

![Captura de tela de 2022-07-04 10-32-04](https://user-images.githubusercontent.com/971483/177165362-8a4a08a3-7499-445c-a87f-f6bae40768f2.png)

leogermani committed:
change the icon to the image icon

leogermani committed:
change label and remove image preview

leogermani committed:
remove isDestructive attribute from button

The red border didnt look great with the border added by focus

leogermani committed:
Merge branch 'trunk' into add/videopress-poster-control

leogermani commented:
I've made some changes:

* Changed the icon
* Removed the preview image. I wasn't happy with the result above, because

*    We were using the full sized image to show the thumbnail, making it sometimes slow and ugly (browser force resizing)
*    We can actually see the image just beneath the menu, in the actual video, so it was redundant.

I added a new screenshot to the description.

pinging @mtias as I think the suggestion to move it from the sidebar to the Toolbar came originally from you. What do you think?

oskosk reviewed (dismissed):
Looks great. Let's merge!

leogermani committed:
Merge branch 'trunk' into add/videopress-poster-control

leogermani committed:
use poster on the saved video url

leogermani commented:
r248904-wpcom
